frontend/src/App.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState} from 'react';
+import ToDoList from './Components/ToDoList'
+import CreateItemForm from './Components/CreateItemForm'
+import {Container,Grid} from '@mui/material';
+import './App.css';
+export default function App() {
+  const [todoItems, setTodoItems] = useState({});
+  const [todoItemsOrderedList,setTodoItemsOrderedList] = useState([]);
+
+  const createToDoItem = (itemName)=>{
+    const newToDo = {
+      itemName : itemName,
+      isCompleted : false,
+      position : 0, 
+    };
+
+    if(Object.values(todoItems).length > 0){
+      newToDo.position = Object.values(todoItems).length
+    }
+    setTodoItems((todoItems) => ({...todoItems, [itemName] : newToDo}));
+  };
+
+  const markToDoItem = (itemName) =>{
+    const newIsCompleted = !todoItems[itemName].isCompleted;
+    setTodoItems((todoItems) => ({...todoItems, [itemName] : {...todoItems[itemName], isCompleted:newIsCompleted }}))
+  }
+
+  const deleteToDoItem = (itemName) =>{
+    const newToDoItems = {...todoItems};
+    delete newToDoItems[itemName];
+    setTodoItems(newToDoItems)
+  } 
+
+  const moveItem = (itemName,dragPosition) => {
+    setTodoItems((todoItems)=>({...todoItems,[itemName] : {...todoItems[itemName], position:dragPosition} }));
+  };
+
+  const sortItems = (items)=>{
+    return items.sort((a, b) => a.position - b.position);
+  }
+
+  const renameTodoItem = (oldItemName,newItemName)=>{
```

Comment:
Continuing the discussion of having an `id` field, then it would make this renaming logic a little less messy. Also as it is currently implemented you cannot allow duplicates, but with an `id` we could allow for duplicates.

Currently, indexing by the text is causing some very buggy behavior